torch/_dynamo/variables/user_defined.py

```diff
@@ -563,3 +563,22 @@ def odict_getitem(self, tx, key):
         )(
             collections.OrderedDict.__getitem__(self.value, key.as_python_constant())
         ).add_options(key, self)
+
+
+class KeyedJaggedTensorVariable(UserDefinedObjectVariable):
+    @staticmethod
+    def is_matching_object(obj):
+        try:
+            from torchrec.sparse.jagged_tensor import KeyedJaggedTensor
+        except ImportError:
+            return False
+        else:
+            return type(obj) is KeyedJaggedTensor
+
+    def __init__(self, value, **kwargs):
+        from torchrec.sparse.jagged_tensor import KeyedJaggedTensor
```

Comment:
This breaks torch.compile if distributed is not build. Can you submit a fix ?

>   File "/scratch/eellison/work/pytorch/torch/_dynamo/convert_frame.py", line 426, in transform
    tracer = InstructionTranslator(
  File "/scratch/eellison/work/pytorch/torch/_dynamo/symbolic_convert.py", line 2018, in __init__
    self.symbolic_locals = collections.OrderedDict(
  File "/scratch/eellison/work/pytorch/torch/_dynamo/symbolic_convert.py", line 2021, in <genexpr>
    VariableBuilder(
  File "/scratch/eellison/work/pytorch/torch/_dynamo/variables/builder.py", line 220, in __call__
    vt = self._wrap(value).clone(**self.options())
  File "/scratch/eellison/work/pytorch/torch/_dynamo/variables/builder.py", line 624, in _wrap
    elif KeyedJaggedTensorVariable.is_matching_object(value):
  File "/scratch/eellison/work/pytorch/torch/_dynamo/variables/user_defined.py", line 572, in is_matching_object
    from torchrec.sparse.jagged_tensor import KeyedJaggedTensor
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/__init__.py", line 8, in <module>
    import torchrec.distributed  # noqa
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/__init__.py", line 36, in <module>
    from torchrec.distributed.model_parallel import DistributedModelParallel  # noqa
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/model_parallel.py", line 19, in <module>
    from torchrec.distributed.planner import (
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/planner/__init__.py", line 22, in <module>
    from torchrec.distributed.planner.planners import EmbeddingShardingPlanner  # noqa
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/planner/planners.py", line 19, in <module>
    from torchrec.distributed.planner.constants import BATCH_SIZE, MAX_SIZE
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/planner/constants.py", line 10, in <module>
    from torchrec.distributed.embedding_types import EmbeddingComputeKernel
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/torchrec/distributed/embedding_types.py", line 14, in <module>
    from fbgemm_gpu.split_table_batched_embeddings_ops import EmbeddingLocation
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/fbgemm_gpu/__init__.py", line 21, in <module>
    from . import _fbgemm_gpu_docs  # noqa: F401, E402
  File "/scratch/eellison/work/torchdynamo/lib/python3.8/site-packages/fbgemm_gpu/_fbgemm_gpu_docs.py", line 18, in <module>
    torch.ops.fbgemm.jagged_2d_to_dense,
  File "/scratch/eellison/work/pytorch/torch/_ops.py", line 748, in __getattr__
    raise AttributeError(
torch._dynamo.exc.InternalTorchDynamoError: '_OpNamespace' 'fbgemm' object has no attribute 'jagged_2d_to_dense'

Reply:
@eellison yeah sure I can submit a fix.

Reply:
@eellison wanna take a look at https://github.com/pytorch/pytorch/pull/107646 ?